### PR TITLE
CSHARP-4781: DefaultIfEmpty using wrong default value.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

File: src/MongoDB.Driver/Support/ReflectionExtensions.cs [hunk 1]
```diff
@@ -25,12 +25,10 @@ internal static class ReflectionExtensions
...
-            if (type.GetTypeInfo().IsValueType)
-            {
-                return Activator.CreateInstance(type);
-            }
-
-            return null;
+            var genericMethod = typeof(ReflectionExtensions)
+                .GetMethod(nameof(GetDefaultValueGeneric), BindingFlags.NonPublic | BindingFlags.Static)
+                .MakeGenericMethod(type);
+            return genericMethod.Invoke(null, null);
```
Comment (rstam) [inline src/MongoDB.Driver/Support/ReflectionExtensions.cs:31]: This change *might* not actually be needed, but I wanted a way to *actually* use `default(TValue)` to be 100% sure I was getting the right default value for all types.

File: tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        public void Select_with_DefaultIfEmpty_should_return_default_int(
+            [Values(LinqProvider.V2, LinqProvider.V3)] LinqProvider linqProvider)
+        {
+            var settings = MongoClientSettings.FromConnectionString("mongodb://localhost");
+            settings.LinqProvider = LinqProvider.V2;
+            var client = new MongoClient(settings);
```
Comment (JamesKovacs) [inline tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs:54]: `settings` and `client` are unused.

File: tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        }
+
+        [Theory]
+        [ParameterAttributeData]
+        public void Select_with_DefaultIfEmpty_should_return_default_string(
+            [Values(LinqProvider.V2, LinqProvider.V3)] LinqProvider linqProvider)
+        {
+            var settings = MongoClientSettings.FromConnectionString("mongodb://localhost");
+            settings.LinqProvider = LinqProvider.V2;
+            var client = new MongoClient(settings);
```
Comment (JamesKovacs) [inline tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs:85]: `settings` and `client` are unused.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Some dead code in tests needs to be removed. As well have we verified that this fix resolves the `$unwind` issue reported in CSHARP-4781?

File: tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs (hunk 2 above)
Comment (rstam) [reply tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs]: Oops. Copy/paste leftovers. Removed.

File: tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs (hunk 3 above)
Comment (rstam) [reply tests/MongoDB.Driver.Tests/Linq/Linq3Implementation/Jira/CSharp4781Tests.cs]: Removed.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**